test/Microsoft.AspNetCore.WebSockets.Test/WebSocketMiddlewareTests.cs

```diff
@@ -558,5 +562,56 @@ public WebSocketMiddlewareTests(ITestOutputHelper output) : base(output)
                 }
             }
         }
+
+        [Theory]
+        [InlineData(HttpStatusCode.OK, null)]
+        [InlineData(HttpStatusCode.Forbidden, "")]
+        [InlineData(HttpStatusCode.Forbidden, "http://e.com")]
+        [InlineData(HttpStatusCode.OK, "http://e.com", "http://example.com")]
+        [InlineData(HttpStatusCode.OK, "*")]
+        [InlineData(HttpStatusCode.OK, "http://e.com", "*")]
+        public async Task CorsIsAppliedToWebSocketRequests(HttpStatusCode expectedCode, params string[] origins)
```

Comment:
nit: **not** CORS! `OriginIsValidatedForWebSocketRequest`.

Might be good to have a parallel test `OriginIsNotValidatedForNonWebSocketRequest` that uses the same Origin headers and AllowedOrigin config, but **is not** a WebSocket request, to validate that the middleware leaves it alone.